src/d3d12/mod.rs

```diff
@@ -740,6 +743,48 @@ impl Allocator {
     }
 }
 
+impl fmt::Debug for Allocator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut allocation_report = vec![];
+
+        for memory_type in &self.memory_types {
+            for block in memory_type.memory_blocks.iter().flatten() {
+                allocation_report.extend_from_slice(&block.sub_allocator.report_allocations())
```

Comment:
Or move the vector without borrowing?

```suggestion
                allocation_report.extend(block.sub_allocator.report_allocations())
```